packages/game-app/src/signup/apis/executeSignup.ts

```diff
@@ -0,0 +1,34 @@
+import { SignupInfo } from '../types/signupInfo';
+import firebase from 'firebase/app';
+import { FirebaseCollection } from '@pipeline/common';
+import 'firebase/auth';
+import 'firebase/firestore';
+import { User } from '../../_shared/auth/slice';
+
+export async function executeSignup(signupInfo: SignupInfo): Promise<User> {
+  try {
+    const credentials = await firebase.auth().createUserWithEmailAndPassword(signupInfo.email, signupInfo.password);
+    if (credentials.user) {
+      const user = credentials.user;
+      await firebase.firestore().doc(`${FirebaseCollection.Users}/${user?.uid}`).set({
+        email: signupInfo.email,
+        role: signupInfo.role,
+        devopsMaturity: signupInfo.devopsMaturity,
+      });
+      return {
+        id: user.uid,
+        email: user.email!,
+      };
+    } else {
+      throw new Error('user-not-created');
+    }
+  } catch (e) {
+    try {
```

Comment:
TODO: add custom firebase errors handling

Reply:
errors are handled and saved into the store using the saga util addRequestStatusmanagement, and then showed in the ui using the hook. Why do we need custom handling for firebase errors? they are in the form {code, message}. I think that the only thing that is missing is the custom wording using i18n, as underlined in the pr description